sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessorEvent.cs

```diff
@@ -4,15 +4,17 @@
 using System;
 using System.Threading.Tasks;
 using Azure.Core;
+using Azure.Messaging.EventHubs.Errors;
 
 namespace Azure.Messaging.EventHubs.Processor
 {
     /// <summary>
-    ///   A <see cref="PartitionEvent" /> that is strongly tied to an <see cref="EventProcessorClient" />.  It
-    ///   provides a means to create event processing checkpoints.
+    ///   Contains information about a partition that has attempted to receive an event from the Azure Event Hub
+    ///   service in an <see cref="EventProcessorClient" /> context, as well as the received event, if any.  It
+    ///   also provides a way of creating a checkpoint based on the information contained in the associated event.
     /// </summary>
     ///
-    public struct EventProcessorEvent
+    public class EventProcessorEvent
```

Comment:
I have a dilemma. This really should be a `struct` but I hate what we would need to do to make it one.   I can't reconcile how we would support mocking without having a public constructor that took a `Func` or other type to allow behavior injection for update checkpoint because `virtual` isn't available.   

I really dislike that, which is why I asked for a `class`.     Should we consider making a `delegate` type or an interface and exposing that `public` to avoid the frequent allocations here?   Design-aside, it would be a non-trivial performance gain.

Thoughts or better ideas?

Reply:
Personally I don't feel strongly about it. I think it's reasonable to have a callback in the constructor, even if used only for mocking. The type has three uses (data, context and checkpoint) and we are simply specifying how it will behave in each case.

I think that's the best we can do. I would just make the `UpdateCheckpointDelegate` take a context and an event data as parameters so we can perfectly mimic an actual `Event Processor` call with the expected arguments.